arbitrator/include/capabilities_interface.tpp

```diff
@@ -23,38 +23,96 @@
+                RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"),
+                            "Could not make a service call to get available strategic service topic names, with detected error: " << error.what() <<". So retrying, attempt no: " << retry_attempt);
+                retry_attempt ++;
+            }
+
+        }
+
+        if (!topics_call_successful)
+        {
+            RCLCPP_ERROR_STREAM(rclcpp::get_logger("arbitrator"),
+                            "Failed to retrieve available strategic plugin service topics! Returning empty maneuver responses");
+            return responses;
+        }
+
+        std::vector<std::string> topics_to_retry = detected_topics;
+        std::vector<std::string> current_topics_to_check;
+        retry_attempt = 0;
+        while (!topics_to_retry.empty() && retry_attempt < MAX_RETRY_ATTEMPTS)
+        {
+            current_topics_to_check = topics_to_retry;
+            topics_to_retry.clear();
+            for (const auto & topic : current_topics_to_check)
+            {
+                try {
+                    using std::literals::chrono_literals::operator""ms;
+
+                    if (registered_strategic_plugins_.find(topic) == registered_strategic_plugins_.end())
+                        registered_strategic_plugins_[topic] = nh_->create_client<carma_planning_msgs::srv::PlanManeuvers>(topic);
+
+                    auto client = registered_strategic_plugins_[topic];
 
-                const auto client = nh_->create_client<carma_planning_msgs::srv::PlanManeuvers>(topic);
-                RCLCPP_DEBUG_STREAM(rclcpp::get_logger("arbitrator"), "found client: " << topic);
+                    if (client->wait_for_service(500ms))
+                        RCLCPP_DEBUG_STREAM(rclcpp::get_logger("arbitrator"), "found client: " << topic);
+                    else
+                    {
+                        topics_to_retry.push_back(topic);
+                        RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"), "Following client timed out: " << topic << ", retrying, attempt no: " << retry_attempt);
+                    }
 
-                const auto response = client->async_send_request(msg);
+                    const auto response = client->async_send_request(msg);
 
-                switch (const auto status{response.wait_for(500ms)}; status) {
-                    case std::future_status::ready:
-                        responses.emplace(topic, response.get());
-                        break;
-                    case std::future_status::deferred:
-                    case std::future_status::timeout:
-                        RCLCPP_WARN_STREAM(rclcpp::get_logger("arbitrator"), "failed...: " << topic);
+                    switch (const auto status{response.wait_for(500ms)}) {
+                        case std::future_status::ready:
+                            responses.emplace(topic, response.get());
+                            break;
+                        default:
```

Comment:
I think having the explicit cases for `deferred` and `timeout` make reading this a little clearer.

Reply:
added both back